compiler/src/iree/compiler/Dialect/Stream/IR/StreamOps.cpp

```diff
@@ -1512,7 +1517,7 @@ LogicalResult TensorCloneOp::verify() {
   // information.
   auto sourceEncoding = llvm::cast<RankedTensorType>(op.getSourceEncoding());
   auto resultEncoding = llvm::cast<RankedTensorType>(op.getResultEncoding());
-  if (sourceEncoding.getEncoding() != resultEncoding.getEncoding()) {
+  if (getEncodingAttr(sourceEncoding) != getEncodingAttr(resultEncoding)) {
```

Comment:
Is this required by `flow.tensor.bitcast` lowering? I.e., the op is lowered to flow.tensor.clone and you need to bypass the check?

I'm not convinced that the change is correct. Because the `getEncodingAttr` checks if the tensor type has `IREE::Encoding::EncodingAttr` attribute. There could be other encodings and it is a bug if we introduce new encodings. E.g., `tensor<3x4xi32, #whatever_other_encoding_with_padding_semantic>` can not be cloned to `tensor<3x4xi32>`. I think we need to have a stronger restriction. Perhaps just relax the check for packed_storage encoding?

https://github.com/iree-org/iree/blob/27e7a9058eded6a20dc098368352329263f77a23/compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.cpp#L278-L280

Reply:
I think you are right. I added extra test for the packed attribute.

Reply:
This is a little tricky. And yes this is required by `flow.tensor.bitcast`, and we would sometimes cast from a tensor without attribute to another tensor with packed attribute. In such case, we shouldn't check if both the source and result matches packed attribute.

I have slightly updated it to exclude packed attribute comparison. Suggestions are welcome.